counter_examples.sh

```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+make clean && make
+
+make run &
+
+# wait until program is ready to listen
+# not sure how else to it now
+sleep 3;
+
+./test/data/counter.sh
+
+pid=$(pgrep pcp-statsd.out)
+kill -USR1 $pid
+# I have no idea how to watch output of parallel task and block until given text is found
+sleep 5
+kill -INT $pid
```

Comment:
You can replace all 5 lines with simple `killall pcp-statsd.out` which sends out TERM by default. Not sure why you send USR1, but the typical signal to terminate gracefuly is SIGTERM (`-TERM` which is the default).

You don't need to wait 5 seconds and then interrupt, not necessary I guess.

Reply:
The sleep is there to wait until procedure printing currently recorded values finishes.

Reply:
I figured out below, thanks.